### PR TITLE
Update to Ape 0.8

### PR DESCRIPTION
**Pull request by derekpierre**
**Type of PR:**
- [ ] Bugfix
- [ ] Feature
- [ ] Documentation
- [x] Other

**Required reviews:** 
- [ ] 1
- [x] 2
- [ ] 3

**What this does:**
> High-level idea of the changes introduced in this PR. 
> List relevant API changes (if any), as well as related PRs and issues.

- Update `ape` dependency to latest version
- Adjust `ape-config.yml` based on `ape` updates in latest version
- Update test account management

**Issues fixed/closed:**
> - Fixes #...

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
> What should reviewers focus on? 
> Is there a particular commit/function/section of your PR that requires more attention from reviewers?

### CODE REVIEW

**Review by manumonti — APPROVED** [review]
LGTM! 👌

**Review by theref — APPROVED** [review]

File: tests/utils/blockchain.py [hunk 1]
```diff
@@ -66,6 +67,24 @@ def accounts(self) -> List[AccountAPI]:
         self.__ape_accounts = test_accounts
         return test_accounts
 
+    @singledispatchmethod
+    def __getitem__(self, account_id):
+        raise NotImplementedError(f"Cannot use {type(account_id)} as account ID.")
+
+    @__getitem__.register
```
Comment (cygnusv) [inline tests/utils/blockchain.py:74]: hmm, what's this?

File: tests/integration/blockchain/test_integration_dkg_ritual.py [hunk 2]
```diff
@@ -380,19 +382,16 @@ def test_ursula_ritualist_good_cohort(
     )
 
 
-@pytest.mark.xfail(reason="This is not fixed yet")
```
Comment (cygnusv) [inline tests/integration/blockchain/test_integration_dkg_ritual.py:383]: 💯

**Review by cygnusv — APPROVED** [review]

File: tests/utils/blockchain.py (hunk 1 above)
Comment (derekpierre) [reply tests/utils/blockchain.py:74]: It's related to `singledispatchmethod` in `functools` - https://docs.python.org/3/library/functools.html#functools.singledispatchmethod - where you can define a generic function with different implementations based on parameter types.

This is something `ape` uses for their `TestAccountManager` that I had to mimic to override. In this case, you can get an `AccountAPI` object using either an int index (e.g. `accounts[0]`) or a string checksum address (e.g. `accounts["0x..."]`).

**Pull request merged**